### PR TITLE
Netctl fixes

### PR DESCRIPTION
**Pull request by erikh**
see commits. Found these while testing with a demo script.

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -118,7 +118,8 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
             end
 
             # mount the host directories
-            node.vm.synced_folder "../../../../", gopath_folder
+            node.vm.synced_folder "bin", File.join(gopath_folder, "bin")
+            node.vm.synced_folder ".", File.join(gopath_folder, "src/github.com/contiv/netplugin")
```
Comment (shaleman) [inline Vagrantfile]: @erikh most of us work on multiple Repos like deploy/libcompose/netplugin/objmodel/ofnet in same workspace.
So, we'd need entire source tree available inside the VM for builds.
Can we mount synced_folder $GOPATH/src as /opt/gopath/src inside VM?
this way, binaries don't conflict between Mac and Linux and we can still have entire source tree inside the VM

Comment (erikh) [inline Vagrantfile]: Sure, seems good. Just not the bins or pkg.

**Comment by erikh** [comment]
@shaleman I incorporated some changes around your suggestions. I also managed to keep the existing behavior for those who are not go devs and will not have a GOPATH configured. I hope this helps!

**Comment by shaleman** [comment]
LGTM

**Pull request merged**